src/native/corehost/hostmisc/pal.windows.cpp

```diff
@@ -276,8 +276,10 @@ bool pal::get_default_installation_dir(pal::string_t* recv)
     }
     //  ***************************
 
+    USHORT pProcessMachine, pNativeMachine;
     const pal::char_t* program_files_dir;
-    if (pal::is_running_in_wow64())
+    auto is_running_in_wow64 = IsWow64Process2(GetCurrentProcess(), &pProcessMachine, &pNativeMachine);
```

Comment:
On top of @jkotas's comment, the function doesn't return true/false (isWow/isnotWow). The return value is `FALSE` only if the function fails. In which case we used to treat it as if it's not a Wow process, so we should probably continue to do so.

The answer is looking at the `pProcessMachine` which should be "unknown" if it's not a Wow process, otherwise it should be some "known" value. The correct version (if the new API is available) is to check that the `pProcessMachine` is x86, and only then use the `ProgramFiles(x86)` value.